### DISCUSSION TITLE
Example Menubar hover

### DISCUSSION THREAD

**Discussion by alekseykurylev** [ariakit/ariakit#3401] [Questions]
@diegohaz Hi! I cannot achieve the correct operation of the Menubar component with a hover. I'm taking an example from the documentation as a basis. https://ariakit.org/components/menubar . But when adding `showOnHover`, the problem of automatic closing appears - this is solved by adding `hideTimeout`, but there is an overlap problem when the cursor moves through the main sections of the menu. Could you help and add an example of a Menubar component with a hover using your main example https://ariakit.org/components/menubar 

https://github.com/ariakit/ariakit/assets/8526127/69f65ab7-6ac2-454c-8121-d1c11a9985cb

**Comment by diegohaz** [comment]
> But when adding `showOnHover`, the problem of automatic closing appears

What's this problem? Could you provide a reproduction?

**Reply by alekseykurylev** [reply]
[codesandbox](https://codesandbox.io/p/sandbox/menubar-hover-forked-8nqwpw?layout=%257B%2522sidebarPanel%2522%253A%2522EXPLORER%2522%252C%2522rootPanelGroup%2522%253A%257B%2522direction%2522%253A%2522horizontal%2522%252C%2522contentType%2522%253A%2522UNKNOWN%2522%252C%2522type%2522%253A%2522PANEL_GROUP%2522%252C%2522id%2522%253A%2522ROOT_LAYOUT%2522%252C%2522panels%2522%253A%255B%257B%2522type%2522%253A%2522PANEL_GROUP%2522%252C%2522contentType%2522%253A%2522UNKNOWN%2522%252C%2522direction%2522%253A%2522vertical%2522%252C%2522id%2522%253A%2522clryxotdh00063d5tigjfpu0e%2522%252C%2522sizes%2522%253A%255B70%252C30%255D%252C%2522panels%2522%253A%255B%257B%2522type%2522%253A%2522PANEL_GROUP%2522%252C%2522contentType%2522%253A%2522EDITOR%2522%252C%2522direction%2522%253A%2522horizontal%2522%252C%2522id%2522%253A%2522EDITOR%2522%252C%2522panels%2522%253A%255B%257B%2522type%2522%253A%2522PANEL%2522%252C%2522contentType%2522%253A%2522EDITOR%2522%252C%2522id%2522%253A%2522clryxotdh00023d5tbat3pq2g%2522%257D%255D%257D%252C%257B%2522type%2522%253A%2522PANEL_GROUP%2522%252C%2522contentType%2522%253A%2522SHELLS%2522%252C%2522direction%2522%253A%2522horizontal%2522%252C%2522id%2522%253A%2522SHELLS%2522%252C%2522panels%2522%253A%255B%257B%2522type%2522%253A%2522PANEL%2522%252C%2522contentType%2522%253A%2522SHELLS%2522%252C%2522id%2522%253A%2522clryxotdh00033d5t7a98lu1s%2522%257D%255D%252C%2522sizes%2522%253A%255B100%255D%257D%255D%257D%252C%257B%2522type%2522%253A%2522PANEL_GROUP%2522%252C%2522contentType%2522%253A%2522DEVTOOLS%2522%252C%2522direction%2522%253A%2522vertical%2522%252C%2522id%2522%253A%2522DEVTOOLS%2522%252C%2522panels%2522%253A%255B%257B%2522type%2522%253A%2522PANEL%2522%252C%2522contentType%2522%253A%2522DEVTOOLS%2522%252C%2522id%2522%253A%2522clryxotdh00053d5t9ucuykly%2522%257D%255D%252C%2522sizes%2522%253A%255B100%255D%257D%255D%252C%2522sizes%2522%253A%255B50%252C50%255D%257D%252C%2522tabbedPanels%2522%253A%257B%2522clryxotdh00023d5tbat3pq2g%2522%253A%257B%2522id%2522%253A%2522clryxotdh00023d5tbat3pq2g%2522%252C%2522tabs%2522%253A%255B%255D%257D%252C%2522clryxotdh00053d5t9ucuykly%2522%253A%257B%2522tabs%2522%253A%255B%257B%2522id%2522%253A%2522clryxotdh00043d5t8n4rzcpn%2522%252C%2522mode%2522%253A%2522permanent%2522%252C%2522type%2522%253A%2522UNASSIGNED_PORT%2522%252C%2522port%2522%253A0%252C%2522path%2522%253A%2522%252F%2522%257D%255D%252C%2522id%2522%253A%2522clryxotdh00053d5t9ucuykly%2522%252C%2522activeTabId%2522%253A%2522clryxotdh00043d5t8n4rzcpn%2522%257D%252C%2522clryxotdh00033d5t7a98lu1s%2522%253A%257B%2522tabs%2522%253A%255B%255D%252C%2522id%2522%253A%2522clryxotdh00033d5t7a98lu1s%2522%257D%257D%252C%2522showDevtools%2522%253Atrue%252C%2522showShells%2522%253Atrue%252C%2522showSidebar%2522%253Atrue%252C%2522sidebarPanelSize%2522%253A15%257D)

For the `MenuProvider` component, I added `hideTimeout={300}` and for the `MenuButton` component, `showOnHover` was added

**Reply by diegohaz** [reply]
The link doesn't seem to work. Also, you mentioned there was a problem with `showOnHover` (without `hideTimeout`). That's the repro I was asking for.

**Reply by alekseykurylev** [reply]
[codesandbox](https://codesandbox.io/p/sandbox/menubar-2g74tg?layout=%257B%2522sidebarPanel%2522%253A%2522EXPLORER%2522%252C%2522rootPanelGroup%2522%253A%257B%2522direction%2522%253A%2522horizontal%2522%252C%2522contentType%2522%253A%2522UNKNOWN%2522%252C%2522type%2522%253A%2522PANEL_GROUP%2522%252C%2522id%2522%253A%2522ROOT_LAYOUT%2522%252C%2522panels%2522%253A%255B%257B%2522type%2522%253A%2522PANEL_GROUP%2522%252C%2522contentType%2522%253A%2522UNKNOWN%2522%252C%2522direction%2522%253A%2522vertical%2522%252C%2522id%2522%253A%2522clr9asqzd00063v6mxhm73cy3%2522%252C%2522sizes%2522%253A%255B70%252C30%255D%252C%2522panels%2522%253A%255B%257B%2522type%2522%253A%2522PANEL_GROUP%2522%252C%2522contentType%2522%253A%2522EDITOR%2522%252C%2522direction%2522%253A%2522horizontal%2522%252C%2522id%2522%253A%2522EDITOR%2522%252C%2522panels%2522%253A%255B%257B%2522type%2522%253A%2522PANEL%2522%252C%2522contentType%2522%253A%2522EDITOR%2522%252C%2522id%2522%253A%2522clr9asqzd00023v6mx88w86x2%2522%257D%255D%257D%252C%257B%2522type%2522%253A%2522PANEL_GROUP%2522%252C%2522contentType%2522%253A%2522SHELLS%2522%252C%2522direction%2522%253A%2522horizontal%2522%252C%2522id%2522%253A%2522SHELLS%2522%252C%2522panels%2522%253A%255B%257B%2522type%2522%253A%2522PANEL%2522%252C%2522contentType%2522%253A%2522SHELLS%2522%252C%2522id%2522%253A%2522clr9asqzd00033v6mtt0e04oj%2522%257D%255D%252C%2522sizes%2522%253A%255B100%255D%257D%255D%257D%252C%257B%2522type%2522%253A%2522PANEL_GROUP%2522%252C%2522contentType%2522%253A%2522DEVTOOLS%2522%252C%2522direction%2522%253A%2522vertical%2522%252C%2522id%2522%253A%2522DEVTOOLS%2522%252C%2522panels%2522%253A%255B%257B%2522type%2522%253A%2522PANEL%2522%252C%2522contentType%2522%253A%2522DEVTOOLS%2522%252C%2522id%2522%253A%2522clr9asqzd00053v6mh13yto2s%2522%257D%255D%252C%2522sizes%2522%253A%255B100%255D%257D%255D%252C%2522sizes%2522%253A%255B50%252C50%255D%257D%252C%2522tabbedPanels%2522%253A%257B%2522clr9asqzd00023v6mx88w86x2%2522%253A%257B%2522tabs%2522%253A%255B%257B%2522id%2522%253A%2522clr9asqzc00013v6mhl6uaaoe%2522%252C%2522mode%2522%253A%2522permanent%2522%252C%2522type%2522%253A%2522FILE%2522%252C%2522filepath%2522%253A%2522%252Fpublic%252Findex.html%2522%252C%2522state%2522%253A%2522IDLE%2522%257D%252C%257B%2522id%2522%253A%2522clr9jyb2n00023v6ngklubqvx%2522%252C%2522mode%2522%253A%2522permanent%2522%252C%2522type%2522%253A%2522FILE%2522%252C%2522initialSelections%2522%253A%255B%257B%2522startLineNumber%2522%253A46%252C%2522startColumn%2522%253A20%252C%2522endLineNumber%2522%253A46%252C%2522endColumn%2522%253A20%257D%255D%252C%2522filepath%2522%253A%2522%252Fsrc%252FApp.tsx%2522%252C%2522state%2522%253A%2522IDLE%2522%257D%255D%252C%2522id%2522%253A%2522clr9asqzd00023v6mx88w86x2%2522%252C%2522activeTabId%2522%253A%2522clr9jyb2n00023v6ngklubqvx%2522%257D%252C%2522clr9asqzd00053v6mh13yto2s%2522%253A%257B%2522tabs%2522%253A%255B%257B%2522id%2522%253A%2522clr9asqzd00043v6m4um9qy9v%2522%252C%2522mode%2522%253A%2522permanent%2522%252C%2522type%2522%253A%2522UNASSIGNED_PORT%2522%252C%2522port%2522%253A0%252C%2522path%2522%253A%2522%252F%2522%257D%255D%252C%2522id%2522%253A%2522clr9asqzd00053v6mh13yto2s%2522%252C%2522activeTabId%2522%253A%2522clr9asqzd00043v6m4um9qy9v%2522%257D%252C%2522clr9asqzd00033v6mtt0e04oj%2522%253A%257B%2522tabs%2522%253A%255B%255D%252C%2522id%2522%253A%2522clr9asqzd00033v6mtt0e04oj%2522%257D%257D%252C%2522showDevtools%2522%253Atrue%252C%2522showShells%2522%253Atrue%252C%2522showSidebar%2522%253Atrue%252C%2522sidebarPanelSize%2522%253A15%257D)

Sorry, this link should work.
